### ISSUE TITLE
SCC - Call With Grant 4/27/18

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#862]
In GitLab by @kdjstudios on Apr 27, 2018, 10:44

Requests HD ticket: http://www.servicedesk.answernet.com/profiles/ticket/2018-04-27-99585/conversation

Missing Email HD ticket: http://www.servicedesk.answernet.com/profiles/ticket/2018-05-01-25445/conversation

Hello Team,

I just had a call with Grant to insure he is ready to process the billing on the 2nd. He said he is going to run their billing on the 2nd through their current program and then use that to find any late fees that need to be added as staged fees in SA Billing and then on the 3rd (Thursday) I will process the billing with Grant. I am going to process the billing with him, because he continues to bring up the issues with the last billing cycle attempt and how we lost all of their data; along with how unimpressed he is with the system continually.

He brought up that their team meet and they have a few more concerns that he will be emailing to us by Monday. I informed him that any additional customization requests will be billable services and not be completed until after they are live on SA Billing. I restated that it has been over a year for the on boarding process and this is due to continual requests for additional work. So we are only going to be able to complete work that is causing issues or discrepancies until they go live. 

He also mentioned:

- That he believes most accounts are missing their email addresses? Can we please check this and confirm. 
- That they need to have a true print an mail; and separate pull bills.
 - Such that the print and mail, is actually set to print and mail invoices(not use a third party software); while the pull bill does virtually the same thing.
 - Basically what I could understand is that he wants to have two groups with the 'pull bills' functionality.
 - I informed him that I would discus this with the team and follow up; but that this does initially sound like a billable service. 
- He mentioned about how the 'Credit Card on File' works and why they are not able to use it yet. 
 - I informed him I would check with our team and confirm, but that I thought we already had this discussion with Garrett and this would be a billable service. As it is building an integration between SA Billing and their Payment Gateway for a CC vault.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 27, 2018, 10:45

@smasih @pchaudhary @tim.traylor @rchauhan 

Hello, Just want to insure everyone reads the above conversation. 

- May we check that the emails were imported correctly to all accounts and customers.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 27, 2018, 15:22

Here is the email sent by Garrett:

> We've been discussing some items and we are not sure how we are going to be successful in the use of this billing platform without these items. 
> 
>  
> 
> 1. Being able to record a payment that is attributed a invoice. Preferably being able to create a "matched ledger" would be ideal, but a area with in the recorded payment that allows us to state which invoice a particular payment is paying for. 
> 
>  
> 
> 2. Payments and invoicing. Invoices, we need a way to separate how customers are being invoiced, our clients either receive our invoices by email, On-line submission or by mail. Having specific  invoice media options is needed. Labeled in the SA platform as such. 
> 
> Print and mail - having the ability to print and mail 
> On-line submission - some accounts we have to take a invoice and up-load them on a web application. 
> Email - emailing to a address on file. 
> Pull-bill - having separate option that allows us to select a customers who we need to examine there bills, we are not looking to sort thru 5-800 invoices to find a dozen accounts whom we need to keep a eye on. 
> Payments, We have around 50 accounts whom pay us by Pre-Autherized Payment(PAP). we need a way to generate a list of these accounts so we go and manually take the funds out of their bank account. Preferably a report that allows us to specify how a customer is paying and then generate a list. This may mean that we need a way to select on each account how that customer is paying, so that we can generate a report on such information. 
> 
>  
> 
> * we also need to get the Credit Card payment area of the billing software up and running otherwise we will need a additional invoice media of credit cards so we have the invoices that we will need to manually take the cc payment * 
> 
>  
> 
> 3. how the ageing of the receivables are caluated, currently your system runs on 30, 60, 90 and 120. we need it to run of 28, 56, 72 and 112. this will accurately show customers the current amount outstanding of their account. this also solves a issue of giving our bank a list of accounts and outstanding funds which is greater than 90 days, of which are Banking Line of Credit is based on. 
> 
>  
> 
> Let me know what you think, and obviously we will need to discuss further, these items are going to help insure our success in the use of this platform.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 27, 2018, 15:23

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Apr 28, 2018, 24:25

@kdjstudios 

> That he believes most accounts are missing their email addresses? Can we please check this and confirm.

If I remember correctly they used to provide us separate email import file, which was not in #813

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 28, 2018, 11:11

@smasih @pchaudhary @tim.traylor @thomas.gentry @rchauhan 

WHAT???!!!! How could your team say the import was completed if a whole file was missing???

Please let us know what file is needed. This is a major issue!!!!!

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Apr 30, 2018, 24:06

@kdjstudios 

They gave it once, with account numbers and their emails.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 30, 2018, 10:46

@tim.traylor @thomas.gentry 

Marking for discussion. The way this was handled I would believe is not acceptable and has now resulted in complaints for the client again, due to failure to correctly import their information. 

@smasih 

What file do you need?!

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 1, 2018, 24:41

@kdjstudios 

> What file do you need?!

They would know, a file with email and account number against them.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 1, 2018, 09:02

@smasih 

What file did you have before? I do not want to ask and receive a different file and have you need to create a whole new import script. Please answer the question.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 1, 2018, 16:03

Email sent to Garrett:

> Hello Garrett,
> 
> First we would like to say thank you for bringing this concerns to our attention now. We have been able to discuss these with our team thoroughly. You first stated that you are concerned with your ability to use SA Billing successfully without these new custom features being implemented. If this is so, we would like to stop the parallel phase, as these requests will need to be completed prior to and save your team from having to perform duplicate work on testing in parrelell. 
> 
> In regards to each feature request the support team is currently working on researching and providing us with a cost and time estimate for each. We hope to have the cost and time estimates available to you within 48 hours of receiving clarification on the requests below. With your approval and priority assignment on each request accordingly; we will then have the support team begin working on them. 
> 
> We do need some more clarification and information on the following requests so the support team can provide you with viable estimates. We have each request listed below. Please be extremely detailed to insure your request is conveyed correctly. 
> 
> 1. The ability to notate on a payment what invoice it is for:
>  - This has already been discussed and the team will be able to provide a cost and time estimate without more information.
> 
> 2. Print and Mail: 
>  - Currently SA Billing already has the ability to print and mail invoices manually through the pull bills functionality. We are needing more information regarding the customization you are requesting in order to provide a cost and time estimate. 
> 
> 3. On-line submission:
>  - This is the first we have ever heard from you or your team about this feature. We will need more information regarding the customization you are requesting in order to provide a cost and time estimate. 
> 
> 4. Email: 
>  - SA Billing already has the ability to email invoices. We will need more information regarding the customization you are requesting in order to provide a cost and time estimate. 
> 
> 5. Pull Bill:
>  - From your description we believe you are requesting to have multiple groups of pull bills functionality? Example to pull bills by office. We will need more information regarding the customization you are requesting in order to provide a cost and time estimate. 
> 
> 6. Default payment type:
>  - From your description we believe you are requesting to have the ability to specify the default payment type for each account. We will need more information regarding the customization you are requesting in order to provide a cost and time estimate. 
> 
> 7. Default payment type report:
>  - From your description we believe you are requesting to have a report available for the default payment type functionality requested above.
> 
> 8. Credit Card Payments:
>  - SA Billing already supports the ability to manually process credit cards through your Moneris provider. 
>  - We believe you are requesting to have the Credit Card information saved on file in a Vault in Moneris and also integrated into SA billing. If this is true, we will be able to provide a cost and time estimate without more information. If not, we will need more information regarding the customization you are requesting to provide a cost and time estimate. 
> 
> 9. Aging calculations through out the full system to be updated from the standard 30,60,90,and 120 to a customized 28,56,72, and 112 day.
>  - This would be custom request for changes in at least the following sections of SA Billing: Invoice layout's aging, Aging reports, AR reports, and more. 
>  - Your request seems counterproductive as you state you want a report to give to your bank a report with a 90 day; which SA Billing currently does. However you are requesting to have the calculations based upon 28,56,72, and 112. We will need more information regarding the customization you are requesting to provide a cost and time estimate. 
> 
> Please provide more information detailing out your custom requests for the SA Billing platform in order for our team to be able to provide you with cost and time estimates accordingly.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 1, 2018, 16:10

Also sent:

> From our conversation on Friday you had mentioned that the accounts and customers did not have all of the email addresses imported. After communicating with our support team they informed me that your team had not provided us with the file containing the email addresses on the last import. Please provide us with this file and we will be sure to have the support team import those. Thank you and please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 1, 2018, 16:10

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 1, 2018, 16:10

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 2, 2018, 24:39

@kdjstudios 

> What file did you have before? I do not want to ask and receive a different file and have you need to create a whole new import script. Please answer the question.

I got the question Kyle but I am not able to recall the file name and the ticket associated with it.
Here is the file (modified file name) we got on the initial import [emails2.csv](/uploads/62aefb747af9be1c495d0e3e421e9dc4/emails2.csv)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 2, 2018, 05:47

> The ability to notate on a payment what invoice it is for:
>
> - This has already been discussed and the team will be able to provide a cost and time estimate without more information.

`estimate 3hr` because it's not simple as it sounds. We have to give the user a list of outstanding invoices in chronological order because it's very unlikely that user will remember invoice numbers

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 2, 2018, 07:29

@kdjstudios 

> Credit Card Payments:
>
> - SA Billing already supports the ability to manually process credit cards through your Moneris provider.
>
> - We believe you are requesting to have the Credit Card information saved on file in a Vault in Moneris and also integrated into SA billing. If this is true, we will be able to provide a cost and time estimate without more information. If not, we will need more information regarding the customization you are requesting to provide a cost and time estimate.

Moneris credit card on Functionality is live, it's just we have not configured it for SCC. For the sake of estimate, it took us almost 1.5 days to perfect the functionality with the batch report and testing.

Initially, they provided us CC details so if required we also have the script which can update listed CC details to given moneris vault.

> Aging calculations through out the full system to be updated from the standard 30,60,90,and 120 to a customized 28,56,72, and 112 day.
>
> - This would be custom request for changes in at least the following sections of SA Billing: Invoice layout's aging, Aging reports, AR reports, and more.
>
> - Your request seems counterproductive as you state you want a report to give to your bank a report with a 90 day; which SA Billing currently does. However you are requesting to have the calculations based upon 28,56,72, and 112. We will need more information regarding the customization you are requesting to provide a cost and time estimate.


Discussed today with @tim.traylor and we are planning to make it configurable. There are three subrequests within this request.  
 * Change in invoice print
 * Aging Report Change
 * Invoice Aging Calculation Change   
`estimate 5-6 hours including testing`

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 2, 2018, 08:39

Summary of estimates so far provided. Once we have estimates for all requests I will provide to SCC. 


- The ability to notate on a payment what invoice it is for:

    - 3 Hours

- Print and Mail:

    - Need estimate

- On-line submission:

    - Need estimate

- Email:

    - Need estimate

- Pull Bill:

    - Need estimate

- Default payment type:

    - Need estimate 

- Default payment type report:

    - Need estimate

- Credit Card Payments:

    - 12 Hours (1.5 Days)

- Aging calculations through out the full system to be updated from the standard 30,60,90,and 120 to a customized 28,56,72, and 112 day.

    - 6 Hours

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 3, 2018, 24:20

@kdjstudios 

for pending estimates, there is more info required for an estimate.



> Credit Card Payments:
>
> - 12 Hours (1.5 Days)

This was already completed #530 withing 12hr duration.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 3, 2018, 10:10

Email sent to Garrett:

> The support team has requested to ask for more detailed information on the following requests. Please see in our original email for the initial request and details. 
> 
> - 2: Print and Mail: We are needing to know what your request is; as SA Billing already has a print and mail feature by using the pull bills. Please detail out exactly what you are requesting SA Billing to perform.
> 
> - 3: On-line submission: We have not received any previous information about this type of Billing. Please detail out exactly how you would like this feature to function. 
> 
> - 4: Email: We are needing to know what your request is; as SA Billing already has the ability to send email invoices. Please detail out exactly what you are requesting SA Billing to perform.
> 
> - 5: Pull Bill: Please clarify and explain in more detail this request. We do not have enough information to provide an estimate. 
> 
> - 6: Default payment type: Please clarify and explain in more detail this request. We do not have enough information to provide an estimate. 
> 
> - 7: Default payment type report: Please clarify and explain in more detail this request. We do not have enough information to provide an estimate. 
> 
> The support team believes they have enough information on requests 1, 8, and 9 in order to create an estimate on time and cost and we will reach out if more information is needed. 
> 
> Please keep in mind, we are putting on hold the parallel phase; until we have your clarification on these requests. I did inform Grant that I will be walking him through the billing cycle briefly again today; however we will not be spending lengthy amounts of time on parallel;until these requests have been handled. This is to ensure your team does not have to spend more time entering duplicate information in the parallel phase as you have stated without these requests you will not be successful in using the SA Billing platform. Thank you and we look forward to your response. Please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 7, 2018, 09:53

Sent to Garrett:

> We hope you were able to enjoy your weekend. As we have not received a response from you; we did want to follow up and reiterate that we are putting on hold the parallel phase; until we have your clarification on these requests. Please let us know if you have any questions or concerns. Thank you and we look forward to your response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 23, 2018, 10:17

Updated Priority to 3 for SCC
